Load all libraries

In [38]:
import pandas as pd
import numpy as np
import xarray as xr
import geopandas as gpd

Read in data

In [6]:
# Read in NetCDF file
landsat = xr.open_dataset('landsat8-2025-02-23-palisades-eaton.nc')

# Read in perimeter data
eaton_perimeter = gpd.read_file('Eaton_Perimeter/Eaton_Perimeter_20250121.shp')
palisades_perimeter = gpd.read_file('Palisades_Perimeter/Palisades_Perimeter_20250121.shp')

# Fire perimeter data exploration

Explore the fire perimeter data and write a brief summary of the information you obtained from the preliminary exploration.

In [37]:
print(eaton_perimeter.crs)
print(eaton_perimeter.dtypes)
print(eaton_perimeter.head(2))
print(eaton_perimeter.shape)

EPSG:3857
OBJECTID         int64
type            object
Shape__Are     float64
Shape__Len     float64
geometry      geometry
dtype: object
   OBJECTID            type    Shape__Are  Shape__Len  \
0         1  Heat Perimeter   2206.265625  270.199719   
1         2  Heat Perimeter  20710.207031  839.204218   

                                            geometry  
0  POLYGON ((-13146936.686 4051222.067, -13146932...  
1  POLYGON ((-13150835.463 4052713.929, -13150831...  
(20, 5)


In [36]:
print(palisades_perimeter.crs)
print(palisades_perimeter.dtypes)
print(palisades_perimeter.head(2))
print(palisades_perimeter.shape)

EPSG:3857
OBJECTID         int64
type            object
Shape__Are     float64
Shape__Len     float64
geometry      geometry
dtype: object
   OBJECTID            type   Shape__Are  Shape__Len  \
0         1  Heat Perimeter  1182.082031  267.101144   
1         2  Heat Perimeter  2222.488281  185.498783   

                                            geometry  
0  POLYGON ((-13193543.302 4032913.077, -13193543...  
1  POLYGON ((-13193524.155 4033067.953, -13193524...  
(21, 5)


The data frames for both fires have the same 5 columns including geometry, with the same data types. Palisades's fire has one extra row. Both data frames are in EPSG:3857, which is a projected crs often used in programs like OpenStreetMap, Google Maps, ArcGIS and ESRI.